src/psyclone/psyGen.py

```diff
@@ -5284,11 +5307,62 @@ def local_symbols(self):
         :return:  List of local symbols.
         :rtype: list of :py:class:`psyclone.psyGen.Symbol`
         '''
-        locals_list = []
-        for symbol in self._symbols.values():
-            if symbol.scope == "local":
-                locals_list.append(symbol)
-        return locals_list
+        return [sym for sym in self._symbols.values() if sym.scope == "local"]
+
+    @staticmethod
+    def indent(count, indent=INDENTATION_STRING):
+        '''
+        Helper function to produce indentation strings.
+
+        :param int count: Number of indentation levels.
+        :param str indent: String representing one indentation level.
+        :return: Complete indentation string.
+        :rtype: str
+        '''
+        result = ""
+        for i in range(count):
+            result += indent
+        return result
+
+    def gen_c_local_variables(self, indent=0):
+        '''
+        Generate C code that defines all local symbols in the Symbol Table.
+
+        :param int indent: Indentation level
+        :return: C languague definition of the local symbols.
+        :rtype: str
+        '''
+        code = ""
+        for symbol in self.local_symbols:
+            code += self.indent(indent) + symbol.gen_c_definition() + ";\n"
+        return code
+
+    def gen_ocl_argument_list(self, indent=0):
```

Comment:
You could mark each of these methods as being abstract (@abc.abstractmethod) but then the class itself is formally abstract and cannot be instantiated and that might not be what we want?

Reply:
I think we don't need a completely abstract class yet as just the usecase of generating OpenCL needs is specialised. For other use cases the generic class would be enough.

Reply:
OK.